web/src/stream_settings_ui.js

```diff
@@ -600,6 +600,10 @@ export function setup_page(callback) {
 
         // show the "Stream settings" header by default.
         $(".display-type #stream_settings_title").show();
+
+        if (!$("#stream_filter input[type='text']").val()) {
+            $("#clear_search_stream_name").css("visibility", "hidden");
+        }
```

Comment:
We can directly use the id of the input `search_stream_name`.

Reply:
> We can directly use the id of the input `search_stream_name`.

Now I have updated the code in whole file to use ```search_stream_name``` at all places where previously ```stream_filter input[type='text']``` was being used.